joblib/numpy_pickle.py

```diff
@@ -1,229 +1,238 @@
-"""
-Utilities for fast persistence of big data, with optional compression.
-"""
+"""Utilities for fast persistence of big data, with optional compression."""
 
 # Author: Gael Varoquaux <gael dot varoquaux at normalesup dot org>
 # Copyright (c) 2009 Gael Varoquaux
 # License: BSD Style, 3 clauses.
 
 import pickle
-import traceback
-import sys
 import os
-import zlib
+import gzip
 import warnings
-import struct
-import codecs
 
-from ._compat import _basestring
-
-from io import BytesIO
+from contextlib import closing
 
-PY3 = sys.version_info[0] >= 3
+from .numpy_pickle_utils import PY3
+from .numpy_pickle_utils import _ZFILE_PREFIX
+from .numpy_pickle_utils import Unpickler, Pickler
+from .numpy_pickle_utils import gzip_file_factory
+from .numpy_pickle_utils import _read_magic, _check_filetype
+from .numpy_pickle_utils import _read_bytes, BUFFER_SIZE
+from .numpy_pickle_compat import load_compatibility
+from .numpy_pickle_compat import NDArrayWrapper, ZNDArrayWrapper
+from ._compat import _basestring
 
-if PY3:
-    Unpickler = pickle._Unpickler
-    Pickler = pickle._Pickler
+###############################################################################
+# Utility objects for persistence.
 
-    def asbytes(s):
-        if isinstance(s, bytes):
-            return s
-        return s.encode('latin1')
-else:
-    Unpickler = pickle.Unpickler
-    Pickler = pickle.Pickler
-    asbytes = str
 
+class NumpyArrayWrapper(object):
+    """An object to be persisted instead of numpy arrays.
 
-def hex_str(an_int):
-    """Converts an int to an hexadecimal string
+    This object is in charge of:
```

Comment:
Maybe you should say something along these lines but with a better phrasing: "this object is used to hack into the pickle machinery and read numpy array data from our custom persistence format"